From f354deaf47f07126b22213f9e9670ff49048e8cb Mon Sep 17 00:00:00 2001
From: Li Jun <jun.li@nxp.com>
Date: Thu, 27 Jul 2017 21:13:19 +0800
Subject: [PATCH 2298/5242] MLK-16013-19 staging: typec: tcpci: Only touch
 target bit when enable vconn

commit  772248ed7a66f671ba6ac6fba13bc99e552ff4b6 from
https://source.codeaurora.org/external/imx/linux-imx.git

We need a read and then write back to avoid touch any other bits when
enable or disable vconn.

Reviewed-by: Peter Chen <peter.chen@nxp.com>
Signed-off-by: Li Jun <jun.li@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/staging/typec/tcpci.c |   10 ++++------
 1 file changed, 4 insertions(+), 6 deletions(-)

diff --git a/drivers/staging/typec/tcpci.c b/drivers/staging/typec/tcpci.c
index ae0aaf4..d7430b2 100644
--- a/drivers/staging/typec/tcpci.c
+++ b/drivers/staging/typec/tcpci.c
@@ -282,12 +282,10 @@ static int tcpci_set_vconn(struct tcpc_dev *tcpc, bool enable)
 			return ret;
 	}
 
-	ret = regmap_write(tcpci->regmap, TCPC_POWER_CTRL,
-			   enable ? TCPC_POWER_CTRL_VCONN_ENABLE : 0);
-	if (ret < 0)
-		return ret;
-
-	return 0;
+	ret = regmap_update_bits(tcpci->regmap, TCPC_POWER_CTRL,
+				TCPC_POWER_CTRL_VCONN_ENABLE,
+				enable ? TCPC_POWER_CTRL_VCONN_ENABLE : 0);
+	return ret;
 }
 
 static int tcpci_set_roles(struct tcpc_dev *tcpc, bool attached,
-- 
1.7.9.5

